dspace-api/src/main/java/org/dspace/content/RelationshipMetadataServiceImpl.java

```diff
@@ -0,0 +1,238 @@
+ */
+package org.dspace.content;
+
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.Logger;
+import org.dspace.content.service.MetadataFieldService;
+import org.dspace.content.service.RelationshipService;
+import org.dspace.content.virtual.VirtualMetadataConfiguration;
+import org.dspace.content.virtual.VirtualMetadataPopulator;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class RelationshipMetadataServiceImpl implements RelationshipMetadataService {
+
+    /**
+     * log4j category
+     */
+    private static final Logger log = org.apache.logging.log4j.LogManager.getLogger();
+
+    @Autowired(required = true)
+    protected RelationshipService relationshipService;
+
+    @Autowired(required = true)
+    protected VirtualMetadataPopulator virtualMetadataPopulator;
+
+    @Autowired(required = true)
+    protected MetadataFieldService metadataFieldService;
+
+    @Override
+    public List<RelationshipMetadataValue> getRelationshipMetadata(Item item, boolean enableVirtualMetadata) {
+        Context context = new Context();
+        List<RelationshipMetadataValue> fullMetadataValueList = new LinkedList<>();
+        try {
+            List<MetadataValue> list = item.getMetadata();
+            String entityType = getEntityTypeStringFromMetadata(list);
+            if (StringUtils.isNotBlank(entityType)) {
+                List<Relationship> relationships = relationshipService.findByItem(context, item);
+                for (Relationship relationship : relationships) {
+                    fullMetadataValueList
+                        .addAll(handleItemRelationship(context, item, entityType, relationship, enableVirtualMetadata));
+                }
+
+            }
+        } catch (SQLException e) {
+            log.error("Lookup for Relationships for item with uuid: " + item.getID() + " caused DSpace to crash", e);
+        }
+        return fullMetadataValueList;
+    }
+
+    private String getEntityTypeStringFromMetadata(List<MetadataValue> list) {
+        for (MetadataValue mdv : list) {
+            if (StringUtils.equals(mdv.getMetadataField().getMetadataSchema().getName(),
+                "relationship")
+                && StringUtils.equals(mdv.getMetadataField().getElement(),
+                "type")) {
+
+                return mdv.getValue();
+            }
+        }
+        return null;
+    }
+
+    //This method processes the Relationship of an Item and will return a list of RelationshipMetadataValue objects
+    //that are generated for this specfic relationship for the item through the config in VirtualMetadataPopulator
+    private List<RelationshipMetadataValue> handleItemRelationship(Context context, Item item, String entityType,
```

Comment:
Another method name that is confusing.  I think the term "handle" in method names is not very descriptive...is this method validating something, is it parsing something?  Plus, the term "handle" is even more confusing as DSpace supports CNRI Handles (and this term keeps bringing those to mind). Could we find better names for  these various `handle*` methods?

If these methods are doing multiple things, like both *validating* and finding something, maybe we should split this into two separate methods.

Should this be called something like:  `findRelationshipMetadataValueForItemRelationship()`?  Or is that method not descriptive enough?

Reply:
I've renamed it to findRelationshipMetadataValueForItemRelationship (the name you suggested is indeed correct)
Combined with the JavaDocs I hope this clarifies the method's purpose